### PR TITLE
Adapt to TableSession

### PR DESCRIPTION
**Pull request by YangYumings**
**doc**：https://laj14weh3dn.feishu.cn/wiki/PTCFwaK3Fi3Kx3kzi1Dcx0QGnEe?from=from_copylink

### CODE REVIEW

File: iotdb-2.0/src/main/java/cn/edu/tsinghua/iot/benchmark/iotdb200/DMLStrategy/TableSessionManager.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      tableSession.close();
+    } catch (IoTDBConnectionException ioTDBConnectionException) {
+      LOGGER.error("Failed to close TableSession");
+      throw new TsdbException(ioTDBConnectionException);
+    }
+  }
+
+  @Override
+  public void createSchemaTemplate(Template template) {
+    // nothing to do
```
Comment (liyuheng55555) [inline iotdb-2.0/src/main/java/cn/edu/tsinghua/iot/benchmark/iotdb200/DMLStrategy/TableSessionManager.java:124]: Would throwing an UnsupportedOperationException be better?

**Review by liyuheng55555 — CHANGES_REQUESTED** [review]

File: iotdb-2.0/src/main/java/cn/edu/tsinghua/iot/benchmark/iotdb200/IoTDB.java [hunk 2]
```diff
@@ -143,32 +143,24 @@ public void close() throws TsdbException {
...
-                .build();
-        metaSession.open(config.isENABLE_THRIFT_COMPRESSION());
-        sessionListMap.put(metaSession, createTimeseries(schemaList));
-        modelStrategy.registerSchema(sessionListMap, schemaList);
+        SessionManager sessionManager;
+        if (config.getIoTDB_DIALECT_MODE() == SQLDialect.TABLE) {
+          sessionManager = new TableSessionManager(dbConfig);
+        } else {
+          sessionManager = new TreeSessionManager(dbConfig);
+          sessionManager.init();
```
Comment (OneSizeFitsQuorum) [inline iotdb-2.0/src/main/java/cn/edu/tsinghua/iot/benchmark/iotdb200/IoTDB.java:153]: just move it to line 155.

File: iotdb-2.0/src/main/java/cn/edu/tsinghua/iot/benchmark/iotdb200/DMLStrategy/TableSessionManager.java (hunk 1 above)
Comment (OneSizeFitsQuorum) [reply iotdb-2.0/src/main/java/cn/edu/tsinghua/iot/benchmark/iotdb200/DMLStrategy/TableSessionManager.java:124]: +1， it will expose our bugs early

**Review by liyuheng55555 — APPROVED** [review]
LGTM, SessionManager is a good abstraction !

File: iotdb-2.0/src/main/java/cn/edu/tsinghua/iot/benchmark/iotdb200/DMLStrategy/SessionStrategy.java [hunk 3]
```diff
@@ -495,15 +470,9 @@ public void cleanup() {
...
-      }
-    } catch (IoTDBConnectionException ioTDBConnectionException) {
-      LOGGER.error("Failed to close session.");
-      throw new TsdbException(ioTDBConnectionException);
-    } finally {
-      service.shutdown();
+    if (sessionManager != null) {
+      sessionManager.close();
     }
+    service.shutdown();
```
Comment (OneSizeFitsQuorum) [inline iotdb-2.0/src/main/java/cn/edu/tsinghua/iot/benchmark/iotdb200/DMLStrategy/SessionStrategy.java:476]: lack of finally?

**Pull request merged**